derp/derp_server.go

```diff
@@ -141,6 +142,8 @@ type Server struct {
 	// because it includes intra-region forwarded packets as the
 	// src.
 	sentTo map[key.Public]map[key.Public]int64 // src => dst => dst's latest sclient.connNum
+
+	ipMap map[netaddr.IP]key.Public
```

Comment:
You'll need to map from `netaddr.IPPort` ... the IP alone isn't unique on the internet. Many users behind a NAT share an IP but have different source ports.

Reply:
Also, I'd name this more about what it represent and not how (don't need to put "map" in the name).

Like `addrKey` or `keyOfAddr`